### PR TITLE
Fix for Incorrect gpu_maps #1990

### PR DESCRIPTION
**Pull request by lee212**
Fix for #1990

### CODE REVIEW

File: src/radical/pilot/agent/lm/jsrun.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def _create_resource_set_file(self, slots, uid, sandbox):
                 rs_str += ' cpu: {%s}'  % cores
                 if gpu_maps:
                     gpus  = ','.join(str(gpu) for gpu in gpu_maps.pop(0))
-                    rs_str += '; gpu: {%s}' % gpus
+                    rs_str += '; gpu: {%s}' % gpus if gpus != '' else ''
                 rs_str += '}\n'
```
Comment (andre-merzky) [inline src/radical/pilot/agent/lm/jsrun.py]: Did you find why the `gpu_maps` contained an empty list?

Comment (andre-merzky) [reply src/radical/pilot/agent/lm/jsrun.py]: @lee212 : ping?

Comment (lee212) [reply src/radical/pilot/agent/lm/jsrun.py]: I wonder what expected `gpu_maps` looks like. For example, `[[]]` was presented but `if gpu_maps:` condition couldn't detect this as an empty which I believe this is a valid case, isn't it?

Comment (lee212) [reply src/radical/pilot/agent/lm/jsrun.py]: I do not find why `gpu_map` is presented with an empty list.

**Comment by andre-merzky** [comment]
Thanks for the fix, but I'll close this in favor of #2021 which addresses this on the scheduler level.  This way other LMs will benefit from the corrected `gpu_map`.

**Pull request closed without merging**